tests/test_requirements.py

```diff
@@ -89,21 +89,27 @@ def test_empty_extras(self):
         req = Requirement("foo[]")
         self._assert_requirement(req, "foo")
 
+    def test_unclosed_extras(self):
+        with pytest.raises(InvalidRequirement) as e:
+            Requirement("foo[")
+        assert "Closing square bracket is missing" in str(e)
+
     def test_url(self):
-        url_section = "@ http://example.com"
```

Comment:
The old specifications should still be tested. Do they cause the right error?

Reply:
I've added a `test_missing_name` to cover this. Yes, it causes the right error.